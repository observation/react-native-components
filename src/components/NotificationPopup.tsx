import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

import { IconClose } from './Icon'
import LargeButton, { LargeButtonProps } from '../components/LargeButton'
import Popup from '../components/Popup'
import textStyle from '../styles/text'
import theme from '../styles/theme'

type NotificationPopupStaticProps = {
  title: string
  message: string
  leftButton: LargeButtonProps
  rightButton?: LargeButtonProps
  onClose?: () => void
}

type NotificationPopupProps = NotificationPopupStaticProps & {
  visible: boolean
}

const NotificationPopup = ({ visible, title, message, leftButton, rightButton, onClose }: NotificationPopupProps) => (
  <Popup visible={visible}>
    <View style={styles.mainContainer}>
      <View style={styles.header}>
        <Text style={styles.title}>{title}</Text>

        {onClose && (
          <TouchableOpacity testID="close" onPress={onClose} activeOpacity={0.5} style={styles.closeButton}>
            <IconClose />
          </TouchableOpacity>
        )}
      </View>
      <View style={styles.body}>
        <Text style={textStyle.body}>{message}</Text>
      </View>

      <View style={styles.footer}>
        <View style={styles.buttonContainer}>
          <LargeButton {...leftButton} style={styles.leftButton} />
        </View>
        <View style={styles.buttonContainer}>
          {rightButton && <LargeButton {...rightButton} style={styles.rightButton} />}
        </View>
      </View>
    </View>
  </Popup>
)

const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: 'white',
    margin: theme.margin.common,
    padding: theme.margin.common,
    borderRadius: theme.margin.common,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    flex: 1,
    ...textStyle.title,
  },
  closeButton: {
    marginLeft: theme.margin.half,
  },
  body: {
    marginTop: theme.margin.common,
    marginBottom: theme.margin.double,
  },
  footer: {
    flexDirection: 'row',
  },
  buttonContainer: {
    width: '50%',
  },
  leftButton: {
    margin: 0,
    marginRight: theme.margin.quarter,
  },
  rightButton: {
    margin: 0,
    marginLeft: theme.margin.quarter,
  },
})

export default NotificationPopup

export type { NotificationPopupProps, NotificationPopupStaticProps }
