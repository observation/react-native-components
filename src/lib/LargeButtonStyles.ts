import { StyleProp, TextStyle, ViewStyle } from 'react-native'

import theme from '../styles/theme'

type LargeButtonStyle = {
  buttonStyle: StyleProp<ViewStyle>
  textStyle: StyleProp<TextStyle>
  iconColor: string
}

const primary: LargeButtonStyle = {
  buttonStyle: {
    backgroundColor: theme.color.primary,
  },
  textStyle: { color: theme.color.white },
  iconColor: theme.color.white,
}

const primaryDisabled: LargeButtonStyle = {
  buttonStyle: {
    backgroundColor: theme.color.greySemi,
  },
  textStyle: { color: theme.color.white },
  iconColor: theme.color.white,
}

const secondary: LargeButtonStyle = {
  buttonStyle: {
    borderColor: theme.color.primary,
    borderWidth: 2,
    backgroundColor: theme.color.white,
  },
  textStyle: { color: theme.color.primary },
  iconColor: theme.color.primary,
}

const secondaryDisabled: LargeButtonStyle = {
  buttonStyle: {
    borderColor: theme.color.greySemi,
    borderWidth: 2,
    backgroundColor: theme.color.white,
  },
  textStyle: { color: theme.color.greySemi },
  iconColor: theme.color.greySemi,
}

const secondaryDanger: LargeButtonStyle = {
  buttonStyle: {
    borderColor: theme.color.error,
    borderWidth: 2,
    backgroundColor: theme.color.white,
  },
  textStyle: { color: theme.color.error },
  iconColor: theme.color.error,
}

export type { LargeButtonStyle }

export { primary, primaryDisabled, secondary, secondaryDisabled, secondaryDanger }
