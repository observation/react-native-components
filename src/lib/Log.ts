import crashlytics from '@react-native-firebase/crashlytics'
import { logger, consoleTransport } from 'react-native-logs'

const crashlyticsTransport = ({ msg }: { msg: string }) => crashlytics().log(msg)

const config = {
  transport: __DEV__ ? consoleTransport : crashlyticsTransport,
  levels: {
    trace: 0,
    debug: 1,
    info: 2,
    warn: 3,
    error: 4,
  },
  printLevel: false,
  printDate: true,
  severity: 'debug',
}

interface Logger {
  debug: (...args: unknown[]) => void
  trace: (...args: unknown[]) => void
  info: (...args: unknown[]) => void
  warn: (...args: unknown[]) => void
  error: (...args: unknown[]) => void
}

const Log: Logger = logger.createLogger(config)

export default Log
