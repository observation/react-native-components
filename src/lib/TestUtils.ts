const mockNavigation: any = {
  navigate: jest.fn(),
  addListener: jest.fn(() => jest.fn()),
  replace: jest.fn(),
  goBack: jest.fn(),
  popToTop: jest.fn(),
  canGoBack: jest.fn(),
  setOptions: jest.fn(),
  getState: jest.fn(() => ({ index: 0 })),
}

const mockNavigationContext: any = {
  isFocused: jest.fn(() => true),
  addListener: jest.fn(() => jest.fn()),
  navigate: jest.fn(),
}

export { mockNavigation, mockNavigationContext }
