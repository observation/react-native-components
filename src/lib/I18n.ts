import { I18nManager } from 'react-native'

import Countries from 'i18n-iso-countries'
import { I18n as I18nJs } from 'i18n-js'
import 'moment/min/locales'
import { Scope, TranslateOptions } from 'i18n-js/typings/typing'
import moment from 'moment'
import * as RNLocalize from 'react-native-localize'

import Log from './Log'

Countries.registerLocale(require('i18n-iso-countries/langs/en.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/nl.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/de.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/es.json'))
Countries.registerLocale(require('i18n-iso-countries/langs/fr.json'))

interface SupportedLanguages {
  en: Object
  nl: Object
  de: Object
  fr: Object
  es: Object
}

interface Api {
  setLanguage(language: string): void
}

/** Wrapper class around i18n-js, adding a few more methods */
class I18n {
  i18nJs = new I18nJs()

  constructor(private translations: SupportedLanguages, private api: Api) {
    this.i18nJs.translations = translations
    this.i18nJs.enableFallback = false
    this.configure()
  }

  setLocale = (locale: string) => {
    Log.debug('i18n:setLocale', locale)
    this.i18nJs.locale = locale
    moment.locale(locale)
    this.api.setLanguage(locale)
  }

  get locale() {
    return this.i18nJs.locale
  }

  /** Handle setting an initial or changed locale */
  configure = () => {
    const fallback = { languageTag: 'en', isRTL: false }
    const languages = Object.keys(this.translations)
    const { languageTag, isRTL } = RNLocalize.findBestAvailableLanguage(languages) || fallback
    Log.debug('i18n:configure', languageTag)
    I18nManager.forceRTL(isRTL) // optional, you might not want to handle RTL
    this.setLocale(languageTag)
  }

  getCountryFromLocale = () => {
    Log.debug('i18n:getCountryFromLocale')

    const fallback = 'NL'
    const locales = RNLocalize.getLocales()
    return locales.length > 0 ? locales[0].countryCode : fallback
  }

  getLanguageFromLocale = () => {
    Log.debug('i18n:getLanguageFromLocale')

    const locales = RNLocalize.getLocales()
    return locales.length > 0 ? locales[0].languageCode : null
  }

  country = (code: string) => Countries.getName(code, this.locale, { select: 'official' })

  t = (key: Scope, options?: TranslateOptions | undefined) => this.i18nJs.t(key, options)
}

export default I18n
export type { Api }
