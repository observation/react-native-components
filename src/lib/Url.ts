import { Linking } from 'react-native'

import Log from '../lib/Log'

export const openUrl = (url: string) =>
  Linking.canOpenURL(url).then((supported) => {
    if (supported) {
      return Linking.openURL(url)
    }
    Log.debug('Url:openUrl does not know how to open link', url)
  })
