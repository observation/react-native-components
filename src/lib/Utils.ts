import { Platform, LayoutAnimation } from 'react-native'

import Log from '../lib/Log'

const safeLayoutAnimation = (tag: string) => {
  Log.debug('Utils:safeLayoutAnimation', tag)
  if (Platform.OS === 'ios') {
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
  }
}

const unsafeLayoutAnimation = (tag: string) => {
  Log.debug('Utils:unsafeLayoutAnimation', tag)
  LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
}

export { safeLayoutAnimation, unsafeLayoutAnimation }
