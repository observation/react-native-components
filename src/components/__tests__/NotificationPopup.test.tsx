import React from 'react'

import { NavigationContext } from '@react-navigation/native'
import { fireEvent, render } from '@testing-library/react-native'

import { mockNavigationContext } from '../../lib/TestUtils'
import NotificationPopup from '../NotificationPopup'

describe('NotificationPopup', () => {
  describe('Rendering', () => {
    test('Only left button', () => {
      const { toJSON } = render(
        <NavigationContext.Provider value={mockNavigationContext}>
          <NotificationPopup
            visible={true}
            title={'Title'}
            message={'Message'}
            leftButton={{ title: 'left button', onPress: () => {} }}
          />
        </NavigationContext.Provider>,
      )

      expect(toJSON()).toMatchSnapshot()
    })

    test('Left and right buttons', () => {
      const { toJSON } = render(
        <NavigationContext.Provider value={mockNavigationContext}>
          <NotificationPopup
            visible={true}
            title={'Title'}
            message={'Message'}
            leftButton={{ title: 'left button', onPress: () => {} }}
            rightButton={{ title: 'right button', onPress: () => {} }}
          />
        </NavigationContext.Provider>,
      )

      expect(toJSON()).toMatchSnapshot()
    })

    test('With close button', () => {
      const { toJSON } = render(
        <NavigationContext.Provider value={mockNavigationContext}>
          <NotificationPopup
            visible={true}
            title={'Title'}
            message={'Message'}
            leftButton={{ title: 'left button', onPress: () => {} }}
            rightButton={{ title: 'right button', onPress: () => {} }}
            onClose={() => {}}
          />
        </NavigationContext.Provider>,
      )

      expect(toJSON()).toMatchSnapshot()
    })
  })

  describe('Interaction', () => {
    test('Click on buttons', async () => {
      // GIVEN
      const buttonFunction1 = jest.fn()
      const buttonFunction2 = jest.fn()
      const onClose = jest.fn()

      const { getByTestId, getByText } = render(
        <NavigationContext.Provider value={mockNavigationContext}>
          <NotificationPopup
            visible={true}
            title={'Title'}
            message={'Message'}
            leftButton={{ title: 'Title 1', onPress: buttonFunction1 }}
            rightButton={{ title: 'Title 2', onPress: buttonFunction2 }}
            onClose={onClose}
          />
        </NavigationContext.Provider>,
      )

      // WHEN
      await fireEvent.press(getByText('Title 1'))
      await fireEvent.press(getByText('Title 2'))
      await fireEvent.press(getByTestId('close'))

      // THEN
      expect(buttonFunction1).toBeCalledTimes(1)
      expect(buttonFunction2).toBeCalledTimes(1)
      expect(onClose).toBeCalledTimes(1)
    })
  })
})
