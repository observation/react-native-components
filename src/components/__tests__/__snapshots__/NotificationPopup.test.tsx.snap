// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationPopup Rendering Left and right buttons 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  style={
    {
      "flex": 1,
    }
  }
  transparent={true}
  visible={true}
>
  <BlurView
    blurAmount={2}
    blurType="light"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "backgroundColor": "hsla(0, 0%, 18.1%, 0.6)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 16,
            "margin": 16,
            "padding": 16,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              {
                "color": "#212121",
                "flex": 1,
                "fontFamily": "Ubuntu",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 28,
              }
            }
          >
            Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 32,
              "marginTop": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#666666",
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
              }
            }
          >
            Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "width": "50%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0066B1",
                  "borderRadius": 4,
                  "height": 32,
                  "justifyContent": "center",
                  "margin": 0,
                  "marginRight": 4,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="touchable-opacity"
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginHorizontal": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#212121",
                        "fontFamily": "Ubuntu",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "bold",
                        "lineHeight": 20,
                        "textAlignVertical": "center",
                      },
                      {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  left button
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "width": "50%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0066B1",
                  "borderRadius": 4,
                  "height": 32,
                  "justifyContent": "center",
                  "margin": 0,
                  "marginLeft": 4,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="touchable-opacity"
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginHorizontal": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#212121",
                        "fontFamily": "Ubuntu",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "bold",
                        "lineHeight": 20,
                        "textAlignVertical": "center",
                      },
                      {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  right button
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`NotificationPopup Rendering Only left button 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  style={
    {
      "flex": 1,
    }
  }
  transparent={true}
  visible={true}
>
  <BlurView
    blurAmount={2}
    blurType="light"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "backgroundColor": "hsla(0, 0%, 18.1%, 0.6)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 16,
            "margin": 16,
            "padding": 16,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              {
                "color": "#212121",
                "flex": 1,
                "fontFamily": "Ubuntu",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 28,
              }
            }
          >
            Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 32,
              "marginTop": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#666666",
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
              }
            }
          >
            Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "width": "50%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0066B1",
                  "borderRadius": 4,
                  "height": 32,
                  "justifyContent": "center",
                  "margin": 0,
                  "marginRight": 4,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="touchable-opacity"
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginHorizontal": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#212121",
                        "fontFamily": "Ubuntu",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "bold",
                        "lineHeight": 20,
                        "textAlignVertical": "center",
                      },
                      {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  left button
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "width": "50%",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`NotificationPopup Rendering With close button 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  style={
    {
      "flex": 1,
    }
  }
  transparent={true}
  visible={true}
>
  <BlurView
    blurAmount={2}
    blurType="light"
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "backgroundColor": "hsla(0, 0%, 18.1%, 0.6)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 16,
            "margin": 16,
            "padding": 16,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              {
                "color": "#212121",
                "flex": 1,
                "fontFamily": "Ubuntu",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "bold",
                "lineHeight": 28,
              }
            }
          >
            Title
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginLeft": 8,
                "opacity": 1,
              }
            }
            testID="close"
          >
            <Icon
              color="#939393"
              name="xmark"
              size={18}
              type="light"
            />
          </View>
        </View>
        <View
          style={
            {
              "marginBottom": 32,
              "marginTop": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#666666",
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
              }
            }
          >
            Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "width": "50%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0066B1",
                  "borderRadius": 4,
                  "height": 32,
                  "justifyContent": "center",
                  "margin": 0,
                  "marginRight": 4,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="touchable-opacity"
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginHorizontal": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#212121",
                        "fontFamily": "Ubuntu",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "bold",
                        "lineHeight": 20,
                        "textAlignVertical": "center",
                      },
                      {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  left button
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "width": "50%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0066B1",
                  "borderRadius": 4,
                  "height": 32,
                  "justifyContent": "center",
                  "margin": 0,
                  "marginLeft": 4,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="touchable-opacity"
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginHorizontal": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#212121",
                        "fontFamily": "Ubuntu",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "bold",
                        "lineHeight": 20,
                        "textAlignVertical": "center",
                      },
                      {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  right button
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
